taichi/transforms/compile_to_offloads.cpp

```diff
@@ -35,13 +35,14 @@ void compile_to_offloads(IRNode *ir,
                          bool verbose,
                          bool grad,
                          bool ad_use_stack,
+                         bool ad_reverse_mode,
                          bool start_from_ast) {
   TI_AUTO_PROF;
 
   auto print = make_pass_printer(verbose, kernel->get_name(), ir);
   print("Initial IR");
 
-  if (grad) {
+  if (grad && ad_reverse_mode) {
```

Comment:
I wonder if ad_reverse_mode=true implies grad=True? Given that forward mode doesn't really need a grad kernel?

Reply:
Yes. I agree that ad_reverse_mode=true implies grad=True. Though we doesn't need a grad kernel for forward mode, an additional argument is required to tell that the kernel requires a autodiff pass ?